optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java

```diff
@@ -177,4 +197,38 @@ public Duration getSolvingDuration() {
         }
         return Duration.ofMillis(endingSystemTimeMillis - startingSystemTimeMillis);
     }
+
+    private class UnlockLockPhaseLifecycleListener implements PhaseLifecycleListener {
+
+        @Override
+        public void solvingStarted(SolverScope solverScope) {
+            solverStatusModifyingLock.unlock();
```

Comment:
Seeing a listener like this makes me thing that this whole locking mechanism needs very good Javadocs so that the future us understand what is going on and when.